decoder/src/lib.rs

```diff
@@ -136,6 +137,15 @@ impl FromStr for Encoding {
     }
 }
 
+impl Encoding {
+    pub const fn recoverable(&self) -> bool {
```

Comment:
nit: naming convention (style?) is to name this `is_recoverable` (or maybe `can_recover`) when the method returns a `bool`

Reply:
```suggestion
    pub const fn can_recover(&self) -> bool {
```